src/resolve/resolved-mdns-browse-services.c

```diff
@@ -0,0 +1,454 @@
+                                                JSON_BUILD_PAIR("domain", JSON_BUILD_STRING(domain)),
+                                                JSON_BUILD_PAIR("interface", JSON_BUILD_INTEGER(ss->ifindex))));
+                if (r < 0)
+                        goto finish;
+
+                r = json_variant_append_array(&array, entry);
+                if (r < 0)
+                        goto finish;
+        }
+
+        /* Check for services removed */
+        LIST_FOREACH(mdns_services, service, ss->mdns_services) {
+
+                if (service->family != owner_family)
+                        continue;
+
+                if (dns_answer_contains(answer,service->rr))
+                        continue;
+
+                r = dns_service_split(service->rr->ptr.name, &name, &type, &domain);
+                if (r < 0)
+                        goto finish;
+
+                mdns_remove_service(ss, service);
+
+                log_debug("Remove from the list  %s || %s || %s || %s || %d",
+                          name,
+                          type,
+                          domain,
+                          ((owner_family == AF_INET) ? "AF_INET":"AF_INET6"),
+                          ss->ifindex);
+
+                r = json_build(&entry,
+                                JSON_BUILD_OBJECT(
+                                                JSON_BUILD_PAIR("add_flag", JSON_BUILD_BOOLEAN(false)),
+                                                JSON_BUILD_PAIR("family", JSON_BUILD_INTEGER(owner_family)),
+                                                JSON_BUILD_PAIR("name", JSON_BUILD_STRING(name)),
+                                                JSON_BUILD_PAIR("type", JSON_BUILD_STRING(type)),
+                                                JSON_BUILD_PAIR("domain", JSON_BUILD_STRING(domain)),
+                                                JSON_BUILD_PAIR("interface", JSON_BUILD_INTEGER(ss->ifindex))));
+                if (r < 0)
+                        goto finish;
+
+                r = json_variant_append_array(&array, entry);
+                if (r < 0)
+                        goto finish;
+        }
+
+
+        if (!json_variant_is_blank_array(array)) {
+                _cleanup_(json_variant_unrefp) JsonVariant *vm = NULL;
+                r = json_build(&vm,
+                                JSON_BUILD_OBJECT(
+                                                JSON_BUILD_PAIR("token", JSON_BUILD_UNSIGNED(ss->token)),
+                                                JSON_BUILD_PAIR("browser_service_data", JSON_BUILD_VARIANT(array))));
+                if (r < 0)
+                        goto finish;
+
+                r = varlink_notify(ss->link,vm);
+                if (r < 0)
+                        goto finish;
+        }
+
+finish:
+        if (r < 0) {
+                log_error_errno(r, "Failed to send browse service reply: %m");
```

Comment:
The error is not only in sending notify. Please say e.g. `"Failed to process received services: %m"`.